JavacUtils.classpathEntriesToFiles() doesn't configure dependent projects

Steps to reproduce:

- import [testproj.zip](https://github.com/user-attachments/files/17460339/testproj.zip) 
- open `proj2/src/Foo.java`

You can see several errors because JavacUtils doesn't add proj1's source folder to the classpath. 

You can also try the following:

- import the [weblaf](https://github.com/mgarin/weblaf) project
- open modules/ui/src/com/alee/extended/label/AbstractSimpleStyledTextContent.java
- hover over `@NotNull`, line 78
It doesn't work because NotNull.java is in the depend project's (weblaf-core) source folder.
You can see the following error in the log:
```
java.lang.IllegalStateException: java.lang.NullPointerException: Cannot invoke "com.sun.tools.javac.code.Type.getTypeArguments()" because "t1" is null
	at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.analyze(JavacTaskImpl.java:383)
	at org.eclipse.jdt.core.dom.JavacBindingResolver.resolve(JavacBindingResolver.java:441)
	at org.eclipse.jdt.core.dom.JavacBindingResolver.resolvePackage(JavacBindingResolver.java:1182)
	at org.eclipse.jdt.core.dom.PackageDeclaration.resolveBinding(PackageDeclaration.java:323)
	at org.eclipse.jdt.core.dom.JavacCompilationUnitResolver.resolveBindings(JavacCompilationUnitResolver.java:457)
	at org.eclipse.jdt.core.dom.JavacCompilationUnitResolver.resolveBindings(JavacCompilationUnitResolver.java:451)
	at org.eclipse.jdt.core.dom.JavacCompilationUnitResolver.toCompilationUnit(JavacCompilationUnitResolver.java:540)
	at org.eclipse.jdt.core.dom.ASTParser.internalCreateASTCached(ASTParser.java:1299)
	at org.eclipse.jdt.core.dom.ASTParser.lambda$1(ASTParser.java:1178)
	at org.eclipse.jdt.internal.core.JavaModelManager.cacheZipFiles(JavaModelManager.java:5692)
	at org.eclipse.jdt.core.dom.ASTParser.internalCreateAST(ASTParser.java:1178)
	at org.eclipse.jdt.core.dom.ASTParser.createAST(ASTParser.java:918)
	at org.eclipse.jdt.internal.ui.javaeditor.codemining.JavaMethodParameterCodeMiningProvider.getCompilationUnitNode(JavaMethodParameterCodeMiningProvider.java:97)
	at org.eclipse.jdt.internal.ui.javaeditor.codemining.JavaMethodParameterCodeMiningProvider.collectLineContentCodeMinings(JavaMethodParameterCodeMiningProvider.java:86)
	at org.eclipse.jdt.internal.ui.javaeditor.codemining.JavaMethodParameterCodeMiningProvider.lambda$0(JavaMethodParameterCodeMiningProvider.java:68)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1812)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1804)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:507)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1458)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2034)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:189)
Caused by: java.lang.NullPointerException: Cannot invoke "com.sun.tools.javac.code.Type.getTypeArguments()" because "t1" is null
	at jdk.compiler/com.sun.tools.javac.code.Types.sideCast(Types.java:4508)
	at jdk.compiler/com.sun.tools.javac.code.Types$7.visitClassType(Types.java:1816)
	at jdk.compiler/com.sun.tools.javac.code.Types$7.visitClassType(Types.java:1710)
	at jdk.compiler/com.sun.tools.javac.code.Type$ClassType.accept(Type.java:1053)
	at jdk.compiler/com.sun.tools.javac.code.Types$DefaultTypeVisitor.visit(Types.java:4920)
	at jdk.compiler/com.sun.tools.javac.code.Types.isCastable(Types.java:1656)
	at jdk.compiler/com.sun.tools.javac.comp.Check.checkCastable(Check.java:660)
	at jdk.compiler/com.sun.tools.javac.comp.Check.checkCastable(Check.java:657)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.visitTypeTest(Attr.java:4177)
	at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCInstanceOf.accept(JCTree.java:2277)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.attribTree(Attr.java:674)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.visitParens(Attr.java:3975)
	at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCParens.accept(JCTree.java:2074)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.attribTree(Attr.java:674)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.attribExpr(Attr.java:720)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.visitIf(Attr.java:2266)
	at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCIf.accept(JCTree.java:1613)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.attribTree(Attr.java:674)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.attribStat(Attr.java:748)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.attribStats(Attr.java:767)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.visitBlock(Attr.java:1452)
	at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCBlock.accept(JCTree.java:1133)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.attribTree(Attr.java:674)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.attribStat(Attr.java:748)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.visitIf(Attr.java:2276)
	at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCIf.accept(JCTree.java:1613)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.attribTree(Attr.java:674)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.attribStat(Attr.java:748)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.attribStats(Attr.java:767)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.visitBlock(Attr.java:1452)
	at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCBlock.accept(JCTree.java:1133)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.attribTree(Attr.java:674)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.attribStat(Attr.java:748)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.visitForeachLoop(Attr.java:1592)
	at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCEnhancedForLoop.accept(JCTree.java:1275)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.attribTree(Attr.java:674)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.attribStat(Attr.java:748)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.attribStats(Attr.java:767)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.visitBlock(Attr.java:1452)
	at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCBlock.accept(JCTree.java:1133)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.attribTree(Attr.java:674)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.attribStat(Attr.java:748)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.visitMethodDef(Attr.java:1241)
	at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCMethodDecl.accept(JCTree.java:957)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.attribTree(Attr.java:674)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.attribStat(Attr.java:748)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.attribClassBody(Attr.java:5638)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.attribClass(Attr.java:5526)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.attribClass(Attr.java:5350)
	at jdk.compiler/com.sun.tools.javac.comp.Attr.attrib(Attr.java:5287)
	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.attribute(JavaCompiler.java:1355)
	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.attribute(JavaCompiler.java:1328)
	at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.analyze(JavacTaskImpl.java:404)
	at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.lambda$analyze$1(JavacTaskImpl.java:379)
	at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.invocationHelper(JavacTaskImpl.java:152)
	at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.analyze(JavacTaskImpl.java:379)
	... 20 more
```

--- a/org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacBindingResolver.java
+++ b/org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacBindingResolver.java
@@ -439,7 +439,7 @@ public class JavacBindingResolver extends BindingResolver {
 				// symbols not already present: analyze
 				try {
 					this.javac.analyze();
-				} catch (IOException e) {
+				} catch (IOException | IllegalStateException e) {
 					ILog.get().error(e.getMessage(), e);
 				}
 			}

--- a/org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/JavacUtils.java
+++ b/org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/JavacUtils.java
@@ -342,8 +342,15 @@ public class JavacUtils {
 								// do nothing
 							}
 							if (moduleDescription == null) {
+								IPath path = referencedJavaProject.getOutputLocation();
+								addPath(referencedJavaProject, path, res);
 								for (IClasspathEntry transitiveEntry : referencedJavaProject.resolveClasspath(referencedJavaProject.getExpandedClasspath()) ) {
-									if (transitiveEntry.isExported() || transitiveEntry.getEntryKind() == IClasspathEntry.CPE_SOURCE) {
+									if (transitiveEntry.getEntryKind() == IClasspathEntry.CPE_SOURCE) {
+										IPath outputLocation = transitiveEntry.getOutputLocation();
+										if (outputLocation != null && select.test(transitiveEntry)) {
+											addPath(referencedJavaProject, outputLocation, res);
+										}
+									} else if (transitiveEntry.isExported()) {
 										toProcess.add(transitiveEntry);
 									}
 								}
@@ -352,21 +359,25 @@ public class JavacUtils {
 					}
 				} else if (select.test(current)) {
 					IPath path = current.getPath();
-					File asFile = path.toFile();
-					if (asFile.exists()) {
-						res.add(asFile);
-					} else {
-						IResource asResource = project.getProject().getParent().findMember(path);
-						if (asResource != null && asResource.exists()) {
-							res.add(asResource.getLocation().toFile());
-						}
-					}
+					addPath(project, path, res);
 				}
 			}
 			return res;
 		} catch (JavaModelException ex) {
 			ILog.get().error(ex.getMessage(), ex);
 			return List.of();
+		}
+	}
+
+		private static void addPath(JavaProject project, IPath path, LinkedHashSet<File> res) {
+		File asFile = path.toFile();
+		if (asFile.exists()) {
+			res.add(asFile);
+		} else {
+			IResource asResource = project.getProject().getParent().findMember(path);
+			if (asResource != null && asResource.exists()) {
+				res.add(asResource.getLocation().toFile());
+			}
 		}
 	}
 

